### PR TITLE
LLM - Analytics Opt In Prompt variants A and B

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

Created the feature flag and the screens for the two variants of the Analytics Opt In Prompt

https://github.com/LedgerHQ/ledger-live/assets/17146928/80a36871-a3b4-4a31-81be-7d6429fa4650

https://github.com/LedgerHQ/ledger-live/assets/17146928/ab3109fa-2d73-4cb8-a9b7-3d49259c7ba8

### ❓ Context

- **JIRA or GitHub link**: [LIVE-10936] [LIVE-10937] [LIVE-10938] [LIVE-10939]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-10936]: https://ledgerhq.atlassian.net/browse/LIVE-10936?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-10937]: https://ledgerhq.atlassian.net/browse/LIVE-10937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10938]: https://ledgerhq.atlassian.net/browse/LIVE-10938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10939]: https://ledgerhq.atlassian.net/browse/LIVE-10939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mle-gall — APPROVED** [review]

File: apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  const screensByVariant = {
+    [LlmAnalyticsOptInPromptVariants.variantA]: {
+      main: AnalyticsOptInPromptMainA,
+      details: AnalyticsOptInPromptDetailsA,
+    },
+    [LlmAnalyticsOptInPromptVariants.variantB]: {
+      main: AnalyticsOptInPromptMainB,
+      details: AnalyticsOptInPromptDetailsB,
+    },
+  };
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx:28]: This is a const that does not depend on params or component context, so we can define it above the component.

File: apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <Stack.Navigator screenOptions={stackNavigationConfig}>
+      <Stack.Screen
+        name={ScreenName.AnalyticsOptInPromptMain}
+        component={screensByVariant[activeVariant].main}
+        options={{ title: "", headerRight: undefined }}
+      />
+      <Stack.Screen
+        name={ScreenName.AnalyticsOptInPromptDetails}
+        component={screensByVariant[activeVariant].details}
+        options={{ title: "", headerRight: undefined }}
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx:45]: Why do we need to set `headerRight` to `undefined`? What is the default value?

File: apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Details.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
+import React, { memo } from "react";
+import styled from "styled-components/native";
+import { Flex, IconsLegacy, Link, Text, Toggle } from "@ledgerhq/native-ui";
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Details.tsx:3]: Please remove unused imports

File: apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Details.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+const OptionRow = styled(Flex).attrs({
+  flexDirection: "row",
+  justifyContent: "space-between",
+  alignItems: "center",
+  backgroundColor: "opacityDefault.c05",
+  borderRadius: 12,
+  width: "100%",
+  padding: 4,
+})``;
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Details.tsx:50]: We have a lot of styled components that are shared between both variant and components, can you reuse only one version please?

File: apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Main.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+            flex={1}
+          />
+        </Flex>
+        <Text variant="small" pt={2} color="neutral.c70" textAlign="center" pb="2">
+          {t("analyticsOptIn.variantA.main.content.infoText.info")}
+        </Text>
+        <Link
+          size="small"
+          type="color"
+          onPress={() => {}}
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/screens/AnalyticsOptInPrompt/variantA/Main.tsx:143]: 🗑️, no?

**Review by KVNLS — CHANGES_REQUESTED** [review]
image
Click here does not have the same design as Figma.
Same here:
image

The close cta is missing on the first page:
image

For variant B we don't have the same indentation + same issue on links:
image

File: apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx (hunk 2 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx:45]: By default the headerRight is a Close button component so we need to set it to undefined in our case

**Comment by cgrellard-ledger** [comment]
> How do you navigate to the second screen on variant B, please?

Done since your comment

File: apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx (hunk 2 above)
Comment (mle-gall) [reply apps/ledger-live-mobile/src/components/RootNavigator/AnalyticsOptInPromptNavigator.tsx]: Fixed (second param of getStackNavigatorConfig to false prevent Close buttons renderings, maybe we should plan to transform the props of getStackNavigatorConfig to an object so it's clearer when using it)

**Review by sshmaxime — APPROVED** [review]

**Review by KVNLS — APPROVED** [review]
Works well, congrats!

**Pull request merged**